packages/common/src/auditlogs/ColumnComponents.js

```diff
@@ -87,24 +88,32 @@ const actorMap = {
   device: 'id'
 };
 
-export const UserDescriptor = (item, index) => <div key={`${item.time}-${index} `}>{item.actor[actorMap[item.actor.type]]}</div>;
+const generateKey = (item, index) => `${item.time}-${index}`;
+
+export const UserDescriptor = (item, index) => <div key={generateKey(item, index)}>{item.actor[actorMap[item.actor.type]]}</div>;
 export const ActionDescriptor = (item, index) => (
-  <div className="uppercased" key={`${item.time}-${index}`}>
+  <div className="uppercased" key={generateKey(item, index)}>
     {item.action}
   </div>
 );
 export const TypeDescriptor = (item, index) => (
-  <div className="capitalized" key={`${item.time}-${index}`}>
+  <div className="capitalized" key={generateKey(item, index)}>
     {auditlogTypes[item.object.type]?.title ?? item.object.type}
   </div>
 );
 export const ChangeDescriptor = (item, index) => {
   const FormatterComponent = mapChangeToContent(item).actionFormatter;
-  return <FormatterComponent key={`${item.time}-${index}`} {...item.object} />;
+  return <FormatterComponent key={generateKey(item, index)} {...item.object} />;
 };
 export const ChangeDetailsDescriptor = (item, index, userCapabilities) => {
   const { component: Comp, accessCheck } = mapChangeToContent(item);
-  const key = `${item.time}-${index}`;
+  const key = generateKey(item, index);
   return accessCheck(userCapabilities) ? <Comp key={key} item={item} /> : <div key={key} />;
 };
-export const TimeWrapper = (item, index) => <Time key={`${item.time}-${index}`} value={item.time} />;
+export const TimeWrapper = (item, index) => <Time key={generateKey(item, index)} value={item.time} />;
+
+export const ViewDetails = (item, index, onIssueSelection = () => {}, allowsExpansion = true, ViewComponent = DetailsIndicator) => (
+  <div className={allowsExpansion ? 'clickable' : ''} key={generateKey(item, index)} onClick={() => allowsExpansion && onIssueSelection(item)}>
+    {allowsExpansion && <ViewComponent />}
+  </div>
+);
```

Comment:
since this is very dependent on the `allowsExpansion` props and that defaults to `true` I would handle this outside (via `allowsExpansion ? <DetailsIndicator /> : <div />` or something similar)